tests/components/stt/test_init.py

```diff
@@ -72,22 +74,24 @@ async def async_process_audio_stream(
         return SpeechResult("test", SpeechResultState.SUCCESS)
 
 
-@pytest.fixture
-def test_provider():
+@pytest.fixture(name="test_provider")
+def get_test_provider() -> TestProvider:
```

Comment:
Renamed to avoid confusion with test functions.

Reply:
We should remove the whole `test` part of this (e.g., `TestProvider` is also a problematic name for the class), instead maybe call it "mock" or something.